src/Altinn.Platform/Altinn.Platform.Storage/IntegrationTest/MessageBoxInstancesControllerTest.cs

```diff
@@ -0,0 +1,248 @@
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Altinn.Platform.Storage.Client;
+using Altinn.Platform.Storage.Configuration;
+using Altinn.Platform.Storage.IntegrationTest.Fixtures;
+using Altinn.Platform.Storage.Models;
+using Altinn.Platform.Storage.Repository;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Storage.Interface.Models;
+using Xunit;
+
+namespace Altinn.Platform.Storage.IntegrationTest
+{
+    /// <summary>
+    /// Test class for InstanceController
+    /// </summary>
+    public class MessageBoxInstancesControllerTest : IClassFixture<PlatformStorageFixture>, IDisposable
+    {
+        private readonly PlatformStorageFixture fixture;
+        private readonly HttpClient client;
+        private InstanceClient instanceClient;
+        private InstanceRepository instanceRepository;
+        private ApplicationClient appClient;
+        private MessageBoxInstanceData testdata;
+        private readonly List<string> appIds;
+        private static DocumentClient _client;
+        private AzureCosmosSettings _cosmosettings = new AzureCosmosSettings()
+        {
+            Collection = "Instance",
+            Database = "ServiceEngine",
+            EndpointUri = "https://localhost:8081",
+            PrimaryKey = "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw=="
+        };
+
+        private readonly string versionPrefix = "/storage/api/v1";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstanceStorageTests"/> class.
+        /// </summary>
+        /// <param name="fixture">the fixture object which talks to the SUT (System Under Test)</param>
+        /// <param name="cosmosettings">Cosmos settings</param>
+        /// <param name="logger">Logger. </param>
+        public MessageBoxInstancesControllerTest(PlatformStorageFixture fixture)
+        {
+            this.fixture = fixture;
+            this.client = this.fixture.Client;
+            this.instanceClient = new InstanceClient(this.client);
+            this.appClient = new ApplicationClient(this.client);
+
+            _client = new DocumentClient(new Uri(_cosmosettings.EndpointUri), _cosmosettings.PrimaryKey, new ConnectionPolicy
+            {
+                ConnectionMode = ConnectionMode.Gateway,
+                ConnectionProtocol = Protocol.Https,
+            });
+
+            testdata = new MessageBoxInstanceData();
+            appIds = testdata.GetAppIds();
+            CreateTestApplications();
+
+        }
+
+        /// <summary>
+        /// Make sure repository is cleaned after the tests is run.
+        /// </summary>
+        public void Dispose()
+        {
+            string requestUri = $"{versionPrefix}/instances?instanceOwnerId={testdata.GetInstanceOwnerId()}";
```

Comment:
is the url right? I see that '/sbl' is missing for messagebox instances

Reply:
As we are just listing all instances for an instance owner to get instanceId it is better to use the standard endpoint as we avoid the mapping to messagebox instance before the list is returned.